### DISCUSSION TITLE
Why is emailVerified hardcoded to null for new user in callback-handler

### DISCUSSION THREAD

**Discussion by turnipdabeets** [nextauthjs/next-auth#9064] [Help]
I'm using the Google Provider which passes back a boolean for `profile.email_verified`. However I noticed that in my DB the emailVerified field is always null no matter if I hardcode it to true as demonstrated below. 

```
GoogleProvider({
      clientId: process.env.NEXT_PUBLIC_GOOGLE_CLIENT_ID || '',
      clientSecret: process.env.GOOGLE_CLIENT_SECRET || '',
      profile(profile) {
        console.log(profile, profile.email_verified);
        return {
          id: profile.sub,
          name: profile.name,
          email: profile.email,
          emailVerified: true,
          image: profile.picture,
        };
      },
    })
 ```
 
I would like my scheme to set `emailVerified` to Boolean with a default as false. However it fails saying "Argument `emailVerified` must not be null.\n'"
 
 ```
 model User {
  id                String       @id @default(cuid())
  createdAt         DateTime     @default(now())
  updatedAt         DateTime     @updatedAt  
  name              String?
  email             String       @unique
  emailVerified     Boolean      @default(false)
  image             String?
  role              UserRole     @default(value: user)
  accounts          Account[]
}
```

After debugging I noticed that the `callback-handler` in next auth core is forcing this value to null. Why? This means I can't change my schema to emailVerified as Boolean. Is there a way around this? What's the reasons for forcing this to null?

> https://github.com/nextauthjs/next-auth/blob/90f2377f2e93d3e3bd8aeaedd8d70336e9089c77/packages/core/src/lib/callback-handler.ts#L219

https://github.com/nextauthjs/next-auth/blob/main/packages/core/src/lib/callback-handler.ts#L219C51-L219C70
```
} else {
      // If the current user is not logged in and the profile isn't linked to any user
      // accounts (by email or provider account id)...
      //
      // If no account matching the same [provider].id or .email exists, we can
      // create a new account for the user, link it to the OAuth account and
      // create a new session for them so they are signed in with it.
      const { id: _, ...newUser } = { ...profile, emailVerified: null }
      user = await createUser(newUser)
    }
 ```

**Comment by goerlitz** [comment]
This looks like a duplicate. see also https://github.com/nextauthjs/next-auth/issues/7218
